src/components/dev-dot-content/mdx-components/mdx-heading-permalink/mdx-heading-permalink.module.css

```diff
@@ -0,0 +1,42 @@
+.root {
+	composes: g-focus-ring-from-box-shadow from global;
+	opacity: 0;
+	border-radius: 6px;
+	padding: 4px;
+	height: fit-content;
+	float: right;
+	margin-left: 8px;
+
+	@media (prefers-reduced-motion: no-preference) {
+		transition: opacity 0.2s ease;
+	}
+
+	& .icon {
+		height: 16px;
+		display: flex;
+		align-items: center;
+		justify-content: center;
+	}
```

Comment:
⛏️ Could we pull this out of the body of `.root` to align with our [Group styles by element, not by state or media type](https://www.notion.so/hashicorp/CSS-Modules-753f2647f7aa4a598e28990d9f0d05a8#839a408c41ca4a609b5cdde0942fa3af) practice?